addons/binding/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/dataexchange/NukiHttpClient.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.nuki.dataexchange;
+
+import java.math.BigDecimal;
+import java.util.concurrent.ExecutionException;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.HttpResponseException;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.openhab.binding.nuki.NukiBindingConstants;
+import org.openhab.binding.nuki.dto.BridgeApiInfoDto;
+import org.openhab.binding.nuki.dto.BridgeApiLockActionDto;
+import org.openhab.binding.nuki.dto.BridgeApiLockStateDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link NukiHttpClient} class is responsible for getting data from the Nuki Bridge.
+ *
+ * @author Markus Katter - Initial contribution
+ */
+public class NukiHttpClient {
+
+    private final static Logger logger = LoggerFactory.getLogger(NukiHttpClient.class);
+
+    private HttpClient httpClient;
+    private Configuration configuration;
+
+    public NukiHttpClient(Configuration configuration) {
+        logger.trace("Instantiating NukiHttpClient({})", configuration);
+        this.configuration = configuration;
+        this.httpClient = new HttpClient();
+        long connectTimeout = NukiBindingConstants.CLIENT_CONNECTTIMEOUT;
+        httpClient.setConnectTimeout(connectTimeout);
+    }
+
+    public void startClient() {
+        logger.trace("Starting HttpClient");
+        try {
+            httpClient.start();
+            logger.trace("Started httpClient[{}]", httpClient);
+        } catch (Exception e) {
```

Comment:
Yes, you are right.
Initially I had a much more detailed exception handling but after working with Jetty HttpClient and the Nuki Bridge API I ended up (after reworking and reworking and ... the code) with this very "basic" exception handling. It just suits my needs as I actually just need the HttpResponseException for further error handling.
I know this is not the very best way. Any more hints/tips are really appreciated!